common/src/main/java/org/apache/nemo/common/dag/DAGBuilder.java

```diff
@@ -169,6 +176,94 @@ public DAGBuilder(final DAG<V, E> dag) {
     return this;
   }
 
+  /**
+   * Use this method when there are no edges to be replaced.
+   * @param edgeToReference edge to reference.
+   * @param edgeToInsert  edge to insert.
+   * @return  itself.
+   */
+  public DAGBuilder<V, E> connectSplitterVertexWithoutReplacing(final E edgeToReference, final E edgeToInsert) {
+    final V src = edgeToInsert.getSrc();
+    final V dst = edgeToInsert.getDst();
+    LOG.error("edgeToReference: {} {} {}", edgeToReference.getId(), edgeToReference.getSrc().getId(),
+      edgeToReference.getDst().getId());
+
+    if (vertices.contains(src) && vertices.contains(dst)) {
+      if (!(edgeToInsert instanceof IREdge)) {
+        return this;
+      }
+
+      if (src instanceof TaskSizeSplitterVertex) {
+        TaskSizeSplitterVertex spSrc = (TaskSizeSplitterVertex) src;
+        IREdge internalEdge = spSrc.getEdgeWithInternalVertex((IREdge) edgeToReference);
+        IREdge newInternalEdge = Util.cloneEdge(internalEdge, internalEdge.getSrc(), (IRVertex) dst);
+        spSrc.mapEdgeWithLoop((IREdge) edgeToInsert, newInternalEdge);
+      }
+      if (dst instanceof TaskSizeSplitterVertex) {
+        TaskSizeSplitterVertex spDst = (TaskSizeSplitterVertex) dst;
+        IREdge internalEdge = spDst.getEdgeWithInternalVertex((IREdge) edgeToReference);
+        LOG.error("internal edge: {} {} {}", internalEdge.getId(), internalEdge.getSrc().getId(),
+          internalEdge.getDst().getId());
+        IREdge newInternalEdge = Util.cloneEdge(internalEdge,
+          (IRVertex) src,
+          internalEdge.getDst());
+        spDst.mapEdgeWithLoop((IREdge) edgeToInsert, newInternalEdge);
+      }
+      incomingEdges.get(dst).add(edgeToInsert);
+      outgoingEdges.get(src).add(edgeToInsert);
+    } else {
+      this.buildWithoutSourceSinkCheck().storeJSON("debug", "errored_ir", "Errored IR");
+      throw new IllegalVertexOperationException("The DAG does not contain"
+        + (vertices.contains(src) ? "" : " [source]") + (vertices.contains(dst) ? "" : " [destination]")
+        + " of the edge: [" + (src == null ? null : src.getId())
+        + "]->[" + (dst == null ? null : dst.getId()) + "] in "
+        + vertices.stream().map(V::getId).collect(Collectors.toSet()));
+    }
+    return this;
+  }
+
+  /**
+   * TaskSizeSplitterVertex only appears in IRDAG.
```

Comment:
Specify a brief action of the method please :)

Reply:
Here too! Thanks.